test/format/format.sh

```diff
@@ -371,41 +392,35 @@ while :; do
 		# If we've run out of time, terminate all running jobs.
 		[[ $elapsed -ge $seconds ]] && {
 			verbose "$name: run timed out at $(date)"
-			stop=1
-			forcequit=1
+			force_quit=1
 		}
 	}
 
 	# Start more jobs.
 	while :; do
 		# Check if we're only running the smoke-tests and we're done.
-		[[ $smoke_test -ne 0 ]] && [[ $smoke_next -ge ${#smoke_list[@]} ]] && stop=1
+		[[ $smoke_test -ne 0 ]] && [[ $smoke_next -ge ${#smoke_list[@]} ]] && quit=1
 	
 		# Check if the total number of jobs has been reached.
-		[[ $total_jobs -ne 0 ]] && [[ $count_jobs -ge $total_jobs ]] && stop=1
+		[[ $total_jobs -ne 0 ]] && [[ $count_jobs -ge $total_jobs ]] && quit=1
 
 		# Check if less than 60 seconds left on any timer. The goal is to avoid killing
 		# jobs that haven't yet configured signal handlers, because we rely on handler
 		# output to determine their final status.
-		[[ $seconds -ne 0 ]] && [[ $(($seconds - $elapsed)) -lt 60 ]] && stop=1
+		[[ $seconds -ne 0 ]] && [[ $(($seconds - $elapsed)) -lt 60 ]] && quit=1
 
-		[[ $stop -ne 0 ]] && break;
+		# Don't create more jobs if we're quitting for any reason.
+		[[ $force_quit -ne 0 ]] || [[ $quit -ne 0 ]] && break;
 
 		# Check if the maximum number of jobs in parallel has been reached.
-		n=`jobs -p | wc -l`
-		[[ $n -ge $parallel_jobs ]] && break
+		[[ $running -ge $parallel_jobs ]] && break
```

Comment:
Could `$running` become obsolete here considering there could be a 10-sec-sleep after `resolve()` call, e.g. some previously running jobs conclude to either success or failure jobs? Or the strategy here is to use the value calculated last time to avoid the overhead of another resolve() call?

Reply:
Those status counters will line up as part of the next resolve() call, probably not a big deal spawning some extra parallel jobs for a certain period of time.

Reply:
Running is the current count of running jobs, so I don't think it can be obsolete, it's how we know how many new jobs to start. Or am I misunderstanding you?

Reply:
I mean the `resolve()` call will calculate how many jobs are running/success/failed, but if the 10-sec-sleep is invoked right after, and then we run into here again checking `$running` value that means we lost the past 10-sec job status update (some jobs running-->success, some running-->failure) without called another `resolve()`?

Reply:
Yes; I don't think that's a problem, though, do you?

We'll loop until we have the right number of processes running (and we won't sleep unless the right number of processes, as calculated based on the results from `resolve()`, are running).